source/Calamari/LaunchTools/NodeExecutor.cs

```diff
@@ -34,63 +35,70 @@ public NodeExecutor(CommonOptions options, IVariables variables, ICommandLineRun
-                                                                                debugMode
-                                                                                ))
+                var pathToNode = variables.Get(instructions.NodePathVariable);
+                var nodeExecutablePath = BuildNodePath(pathToNode);
+                var parameters = BuildParams(instructions, variableFile.FilePath);
+                var runningDeployment = new RunningDeployment(variables);
+                var commandLineInvocation = new CommandLineInvocation(nodeExecutablePath, parameters)
                 {
                     WorkingDirectory = runningDeployment.CurrentDirectory,
                     OutputToLog = true,
                     EnvironmentVars = ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables().ToDictionary(e => e.Key, e => e.Value)
                 };
 
                 var commandResult = commandLineRunner.Execute(commandLineInvocation);
-
                 return commandResult.ExitCode;
             }
         }
 
         static string BuildNodePath(string pathToNode) => CalamariEnvironment.IsRunningOnWindows ? Path.Combine(pathToNode, "node.exe") : Path.Combine(pathToNode, "bin", "node");
 
-        static string BuildArgs(string pathToBootstrapper,
-                                string pathToStepPackage,
-                                string pathToSensitiveVariables,
-                                string sensitiveVariablesSecret,
-                                string salt,
-                                string inputsKey,
-                                string deploymentTargetInputsKey,
-                                bool debugMode) =>
-            $"{(debugMode ? "--inspect-brk " : "" )}\"{pathToBootstrapper}\" \"{pathToStepPackage}\" \"{pathToSensitiveVariables}\" \"{sensitiveVariablesSecret}\" \"{salt}\" \"{inputsKey}\" \"{deploymentTargetInputsKey}\"";
+        string BuildParams(NodeInstructions instructions, string sensitiveVariablesFilePath)
+        {
+            var parameters = new List<string>();
+            var debugMode = variables.GetFlag(DebugVariableName, false);
+            if (debugMode)
+            {
+                parameters.Add("--inspect-brk");
+            }
+
+            var pathToBootstrapperFolder = variables.Get(instructions.BootstrapperPathVariable);
+            var pathToBootstrapper = Path.Combine(pathToBootstrapperFolder, "bootstrapper.js");
+            parameters.Add(pathToBootstrapper);
+            parameters.Add(instructions.BootstrapperInvocationCommand);
+            var pathToStepPackage = variables.Get(instructions.TargetPathVariable);
+            parameters.Add(pathToStepPackage);
+            parameters.Add(sensitiveVariablesFilePath);
+            parameters.Add(options.InputVariables.SensitiveVariablesPassword);
+            parameters.Add(AesEncryption.SaltRaw);
+            if (instructions.BootstrapperInvocationCommand == "Execute")
```

Comment:
Should we be explicit here instead of falling back to discovery? i.e. pattern match "Execute" and "Discover" (or whatever the discovery command is), or throw an exception if we don't recognise the command?